controllers/keda/hpa.go

```diff
@@ -172,6 +182,17 @@ func (r *ScaledObjectReconciler) updateHPAIfNeeded(ctx context.Context, logger l
 	return nil
 }
 
+// deleteAndCreateHpa delete old HPA and create new one
+func (r *ScaledObjectReconciler) renameHPA(ctx context.Context, logger logr.Logger, scaledObject *kedav1alpha1.ScaledObject, foundHpa *autoscalingv2beta2.HorizontalPodAutoscaler, gvkr *kedav1alpha1.GroupVersionKindResource) error {
+	logger.Info("Deleting old HPA", "HPA.Namespace", scaledObject.Namespace, "HPA.Name", foundHpa.Name)
+	if err := r.Client.Delete(ctx, foundHpa); err != nil {
```

Comment:
the problem is that the current behavior  in this usecase is that on step 2, KEDA already changes the labels so it won't be enough.  
IMHO this is the right behavior, i.e. KEDA operator should take ownership on existing HPA and change it / delete and re-create it if necessary  
This behavior together with this new feature will ease developers running k8s with HPA to migrate to KEDA without having to delete manually the "old" HPAs they had

Reply:
> the problem is that the current behavior  in this usecase is that on step 2, KEDA already changes the labels so it won't be enough.  
> IMHO this is the right behavior, i.e. KEDA operator should "take ownership" on existing Hpa with same expected name and change it / delete and re-create it if necessary  
> This behavior together with this new feature will ease developers running k8s with HPA to migrate to KEDA without having to delete manually the "old" HPAs they had

Reply:
@zroubalik , let me know what you think

Reply:
@aviadlevy I can see your point, but I really don't we (KEDA) should manage resources it doesn't own.

We can also check owner reference on the HPA.

Reply:
If this is the expected behavior, we need first to change the current behavior.  
Currently when:  
1. setting HPA named `keda-hpa-<soName>`
2. setting ScaledObject named `<soName>`
3. KEDA is taking control on the previous created HPA, adding all the labels and also adding the ownerReference.  

**Further more:**

4. when deleting the scaled object -> HPA is deleted as well

So there are some I/S changes need to be done in the current behavior to match the behavior you expecting

@zroubalik , Do you think this PR is blocked until the current behavior is changed?

Reply:
What I described before is a case where the user (not KEDA) created the HPA first with the name KEDA will use, and then the scaled object created, which mean that currently (without this PR) KEDA is already changing and managing the user resources (and therefore checking the owner reference won't help here as well)

Reply:
Ah gotcha. Atm KEDA doesn't care about ownership and replaces the HPA. I think that we should eventually iterate to state where KEDA checks the HPA ownership.

It might be in a separate PR though. Do you think you could try to tackle this as well? It would be really great.

Reply:
I agree it should be on a different PR. Not sure yet if I can take it (but I'll know in the upcoming days).  

The question is whether we must first fix this or since there is no degradation we can merge this PR?

Also, can you please open an issue describing what need to be done regarding the hpa ownership? I think you'll describe what needs to be done better than me